### PR TITLE
Jon J ruby objects diagnostic

### PR DESCRIPTION
**Pull request by modularjon**
comfort: 4
clarity: 3

used old lessons and ruby documentation

### CODE REVIEW

**Comment by modularjon** [comment]
Oh crap. On question 7, it shouldn't be self.the_method, it should return self at the end of the method.

def move(direction)
  @moves << 'move #{direction}'
  self
end

File: lib/diagnostic.rb [hunk 1]
```diff
@@ -13,7 +13,17 @@
...
+  attr_accessor :name
+  attr_reader :age
+  attr_writer :location
+
+  def initialize(name, age, location)
+    @name = name
+    @age = age
+    @location = location
+  end
+end
```
Comment (berziiii) [inline lib/diagnostic.rb:26]: Correct.

File: lib/diagnostic.rb [hunk 2]
```diff
@@ -22,8 +32,9 @@
 # Finally, assign the modified Person to `Answer.dave` below.
 
 ##
-# your answers here
-Answer.dave = nil
+dave = Person.new('Dave', 32, 'Ohio')
+dave.location = 'Somerville'
+Answer.dave = dave
```
Comment (berziiii) [inline lib/diagnostic.rb:37]: Correct.

File: lib/diagnostic.rb [hunk 3]
```diff
@@ -32,7 +43,11 @@
 # which returns the string "think think think".
 
 ##
-# your code here
+class Developer < Person
+  def solve_problems
+    'think think think'
+  end
+end
```
Comment (berziiii) [inline lib/diagnostic.rb:50]: Correct.

File: lib/diagnostic.rb [hunk 4]
```diff
@@ -66,9 +81,12 @@ def initialize
 end
 
 ##
-# your answers here
-Answer.housecat_noise = nil
-#
+say_hi = "I am a HouseCat, and I go 'meow'"
```
Comment (berziiii) [inline lib/diagnostic.rb:84]: Correct

File: lib/diagnostic.rb [hunk 5]
```diff
@@ -91,7 +109,13 @@ def eat_meat(food)
...
 
 ##
-# your code here
+class Lion < Cat
+  include Carnivorous
+
+  def roar
+    puts 'ROAR!'
+  end
+end
```
Comment (berziiii) [inline lib/diagnostic.rb:118]: Correct.

File: lib/diagnostic.rb [hunk 6]
```diff
@@ -100,9 +124,10 @@ def eat_meat(food)
 # Write your answer as a comment in the section below.
 
 ##
-# your answer, in comments, here
-#
-#
+# A class can only inherit from one other class, whereas a class can include as
+# many modules as needed. Mixins allow easier incorporation of future functionality,
+# and prevents minor changes from having major consequences, often the case in
+# using inheritance
```
Comment (berziiii) [inline lib/diagnostic.rb:130]: Correct.

**Comment by berziiii** [comment]
6 / 6. Nice work.

**Pull request closed without merging**